recipes-kernel/linux-firmware-rpidistro/linux-firmware-rpidistro_git.bb

```diff
@@ -8,6 +8,7 @@ RPi-Distro obtains these directly from Cypress; they are not submitted \
 to linux-firmware for general use."
 HOMEPAGE = "https://github.com/RPi-Distro/firmware-nonfree"
 SECTION = "kernel"
+COMPATIBLE_MACHINE = "^ras"
```

Comment:
Why is the pattern abbreviated? I think it is too short and will affect unexpected MACHINEs.

Reply:
@hiraku-wfs  Other recipes e.g.([rpi-bootfiles.bb](https://github.com/miraclelinux/meta-emlinux/blob/warrior/recipes-bsp/bootfiles/rpi-bootfiles.bb#L16), [rpi-u-boot-scr.bb](https://github.com/miraclelinux/meta-emlinux/blob/warrior/recipes-bsp/rpi-u-boot-scr/rpi-u-boot-scr.bb)) uses this pattern.

Is it okay to change other recipes to "raspberrypi 3-64" too?

Reply:
We should use SOC_FAMILY here like meta-raspberrypi.

https://github.com/agherzan/meta-raspberrypi/blob/0135a02ea577bd39dd552236ead2c5894d89da1d/conf/machine/include/rpi-base.inc
```
SOC_FAMILY = "rpi"
include conf/machine/include/soc-family.inc
```

Please add the above description into the meta-emlinux's raspberrypi3-64.conf, then we can use "^rpi$" pattern for the COMPATIBLE_MACHINE.

Reply:
@hiraku-wfs Thank you the comment. I've updated recipes.

Can build core-image-minimal for rpi.

```
masami@ubuntu1804:~/emlinux/build$ bitbake core-image-minimal
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:39
Parsing of 1338 .bb files complete (0 cached, 1338 parsed). 2356 targets, 69 skipped, 4 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-18.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "raspberrypi3-64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.5"
TUNE_FEATURES        = "aarch64 cortexa53 crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "warrior:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "warrior:aecbfc0d5c84cb48c4f08dfa8534442631b7fd8a"
meta-debian-extended = "warrior:bbb54b3d23659529b106993a87f73493c49468ef"
meta-emlinux         = "update-rpi-recipe:36b4bca01f42feecb53c789879eaa7dcba1bdf7f"

Initialising tasks: 100% |#####################################################################################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 853 Found 0 Missed 853 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2942 tasks of which 5 didn't need to be rerun and all succeeded.
```

Cannot build raspberry pi specific recipes.

```
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake rpi-bootfiles -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 841 entries from dependency cache.
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:17
Parsing of 1338 .bb files complete (452 cached, 886 parsed). 2356 targets, 78 skipped, 4 masked, 0 errors.
ERROR: Nothing PROVIDES 'rpi-bootfiles'
rpi-bootfiles was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake rpi-u-boot-scr -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 2356 entries from dependency cache.
ERROR: Nothing PROVIDES 'rpi-u-boot-scr'
rpi-u-boot-scr was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake linux-firmware-rpidistro -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 2356 entries from dependency cache.
ERROR: Nothing PROVIDES 'linux-firmware-rpidistro'
linux-firmware-rpidistro was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
```